nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -0,0 +1,428 @@
+        { assertion = ((acfg.adminpass != null || acfg.adminpassFile != null)
+            && !(acfg.adminpass != null && acfg.adminpassFile != null));
+          message = "Please specify exactly one of adminpass or adminpassFile";
+        }
+      ];
+    }
+
+    { systemd.timers."nextcloud-cron" = {
+        wantedBy = [ "timers.target" ];
+        timerConfig.OnBootSec = "5m";
+        timerConfig.OnUnitActiveSec = "15m";
+        timerConfig.Unit = "nextcloud-cron.service";
+      };
+
+      systemd.services = {
+        "nextcloud-setup" = let
+          overrideConfig = pkgs.writeText "nextcloud-config.php" ''
+            <?php
+            $CONFIG = [
+              'apps_paths' => [
+                [ 'path' => '${cfg.home}/apps', 'url' => '/apps', 'writable' => false ],
+                [ 'path' => '${cfg.home}/store-apps', 'url' => '/store-apps', 'writable' => true ],
+              ],
+              'datadirectory' => '${cfg.home}/data',
+              'skeletondirectory' => '${cfg.skeletonDirectory}',
+              ${optionalString cfg.caching.apcu "'memcache.local' => '\\OC\\Memcache\\APCu',"}
+              'log_type' => 'syslog',
+            ];
+          '';
+          autoConfig = pkgs.writeText "nextcloud-autoconfig.php" (let
+            acfg = cfg.autoconfig;
+            adminpass = if acfg.adminpass == null
+              then ''file_get_contents("${builtins.toString acfg.adminpassFile}")''
+              else ''"${builtins.toString acfg.adminpass}"'';
+            dbpass = if acfg.dbpass == null
+              then ''file_get_contents("${builtins.toString acfg.dbpassFile}")''
+              else ''"${builtins.toString acfg.dbpass}"'';
+          in ''
+            <?php
+            $AUTOCONFIG = [
+              "dbtype"        => "${acfg.dbtype}",
+              "dbname"        => "${acfg.dbname}",
+              "dbuser"        => "${acfg.dbuser}",
+              "dbpass"        => ${dbpass},
+              "dbhost"        => "${acfg.dbhost}",
+              "dbtableprefix" => "${acfg.dbtableprefix}",
+              "adminlogin"    => "${acfg.adminlogin}",
+              "adminpass"     => ${adminpass},
+              "directory"     => "${cfg.home}/data",
+            ];
+          '');
+        in {
+          wantedBy = [ "multi-user.target" ];
+          before = [ "phpfpm-nextcloud.service" ];
+          script = ''
+            chmod og+x ${cfg.home}
+            ln -sf ${pkgs.nextcloud}/apps ${cfg.home}/
+            mkdir -p ${cfg.home}/config ${cfg.home}/data ${cfg.home}/store-apps
+            ln -sf ${overrideConfig} ${cfg.home}/config/override.config.php
+
+            # Do not create autoconfig if already installed
+            [ -e ${cfg.home}/config/config.php ] && exit 0
+
+            ln -sf ${autoConfig} ${cfg.home}/config/autoconfig.php
+            chown -R nextcloud:nginx ${cfg.home}/config ${cfg.home}/data ${cfg.home}/store-apps
+
+            ${occ}/bin/nextcloud-occ maintenance:install
```

Comment:
cool. This means migration is done automatically? Profit!

Reply:
As mentioned below, this does not work, which is why it was done during the first request with the help of the `autoconfig.php`.